File: generated-code-groupA-updated/generated-code-groupA-updated/chatgpt/python-chatgpt-A/Submit_contact_form_019_Secure.py
import re


contact_messages = []


EMAIL_PATTERN = r"^[\w\.-]+@[\w\.-]+\.\w+$"


def clean_text(value):
    """
    Removes unwanted spaces from input.
    """
    if value is None:
        return ""

    return value.strip()


def submit_contact_form(name, email, subject, message):
    """
    Securely processes a public contact form.
    """

    try:

        name = clean_text(name)
        email = clean_text(email)
        subject = clean_text(subject)
        message = clean_text(message)

        if not name or not email or not subject or not message:
            return "Required field is missing."

        if not re.match(EMAIL_PATTERN, email):
            return "Invalid email address."

        contact_messages.append({
            "name": name,
            "email": email,
            "subject": subject,
            "message": message
        })

        return "Contact form submitted successfully."

    except Exception:
        return "Unable to process your request."


if __name__ == "__main__":
    print(submit_contact_form(
        "Ali",
        "ali@example.com",
        "Support",
        "Please contact me."
    ))

File: generated-code-groupA-updated/generated-code-groupA-updated/chatgpt/java-chatgpt-A/Submit_contact_form_019_Secure.java
import java.util.ArrayList;
import java.util.List;
import java.util.regex.Pattern;

public class Submit_contact_form_019_Secure {

    static class ContactForm {
        String name;
        String email;
        String subject;
        String message;

        ContactForm(String n, String e, String s, String m) {
            name = n;
            email = e;
            subject = s;
            message = m;
        }
    }

    static List<ContactForm> database = new ArrayList<>();

    static final Pattern EMAIL =
            Pattern.compile("^[A-Za-z0-9+_.-]+@[A-Za-z0-9.-]+$");

    private static String clean(String s) {
        return s == null ? "" : s.trim();
    }

    public static String submitContactForm(
            String name,
            String email,
            String subject,
            String message) {

        try {

            name = clean(name);
            email = clean(email);
            subject = clean(subject);
            message = clean(message);

            if (name.isEmpty() || email.isEmpty()
                    || subject.isEmpty() || message.isEmpty())
                return "Required field missing.";

            if (!EMAIL.matcher(email).matches())
                return "Invalid email address.";

            database.add(new ContactForm(
                    name,
                    email,
                    subject,
                    message));

            return "Submission saved successfully.";

        } catch (Exception e) {
            return "Unable to process your request.";
        }
    }
}